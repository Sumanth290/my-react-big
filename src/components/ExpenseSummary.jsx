import React from "react";
import { connect } from "react-redux";
import getVisibleExpenses from "../selectors/expenses";
import getExpensesTotals from "../selectors/expenses-total";
import myFormat from "../util/amtFormatter";

export class ExpenseSummary extends React.Component {
    render(){
        return (
            <div>
                {this.props.count > 0 && (
                    this.props.count === 1 ? (
                        <p>{`Viewing 1 expense totalling ₹ ${myFormat(this.props.total)}`}</p>
                    ) : (
                        <p>{`Viewing ${this.props.count} expenses totalling ₹ ${myFormat(this.props.total)}`}</p>
                    )
                )}
            </div>
        );
    };
}

const mapStateToProps = (state) => {
    const visibleExpenses = getVisibleExpenses(state.expenses,state.filters);
    return ({
        count : visibleExpenses.length,
        total : getExpensesTotals(visibleExpenses)/100
    });
};


export default connect(mapStateToProps)(ExpenseSummary);