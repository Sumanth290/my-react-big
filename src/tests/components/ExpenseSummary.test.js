import React from "react";
import {shallow} from "enzyme";
import {ExpenseSummary} from "../../components/ExpenseSummary.jsx";

test("Should render expense Summary for one expense",()=>{
    const wrapper = shallow(<ExpenseSummary count={1} total={234.54}/>)
    expect(wrapper).toMatchSnapshot();
});

test("Should render expense Summary for multiple expenses",()=>{
    const wrapper = shallow(<ExpenseSummary count={5} total={23453245.54}/>)
    expect(wrapper).toMatchSnapshot();
});