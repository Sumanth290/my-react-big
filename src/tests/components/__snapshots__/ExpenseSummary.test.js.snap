// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render expense Summary for multiple expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <div
      className="page-header__content"
    >
      <p
        className="page-header__title"
      >
        Viewing 
        <span>
          5
        </span>
         
        expenses
         totalling 
        <span>
          ₹ 
          2,34,53,245.54
        </span>
      </p>
      <div
        className="page-header__action"
      >
        <Link
          className="buttons"
          replace={false}
          to="/create"
        >
          Create Expense
        </Link>
      </div>
    </div>
  </div>
</div>
`;

exports[`Should render expense Summary for one expense 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <div
      className="page-header__content"
    >
      <p
        className="page-header__title"
      >
        Viewing 
        <span>
          1
        </span>
         
        expense
         totalling 
        <span>
          ₹ 
          234.54
        </span>
      </p>
      <div
        className="page-header__action"
      >
        <Link
          className="buttons"
          replace={false}
          to="/create"
        >
          Create Expense
        </Link>
      </div>
    </div>
  </div>
</div>
`;
